### PR TITLE
Associating Spans with their TracerProvider

### PR DESCRIPTION
**Pull request by zsistla**
* Added functionality to associate a traceprovider resource on a tracer and a tracer resource with a span.

* Added a `defaultResource` method to `ResourceInfo`.

* Added test cases to ‘ResourceTest.php`, `TracerProviderTest.php`, `TracingTest.php`.

### CODE REVIEW

File: sdk/Trace/Span.php [hunk 1]
```diff
@@ -62,6 +70,11 @@ public function __construct(
         $this->events = new Events();
     }
 
+    public function getResource(): ResourceInfo
+    {
+        return clone $this->resource;
```
Comment (bobstrecansky) [inline sdk/Trace/Span.php:75]: What's the reason behind returning a clone of `$this->resource`?

File: sdk/Trace/Tracer.php [hunk 2]
```diff
@@ -167,7 +168,10 @@ public function getSpans(): array
     {
         return $this->spans;
     }
-
+    public function getResource(): ResourceInfo
+    {
+        return clone $this->resource;
```
Comment (bobstrecansky) [inline sdk/Trace/Tracer.php:173]: same question in re: `clone`

**Review by bobstrecansky — APPROVED** [review]
Quick question but not a blocker.  Awesome work @zsistla !

File: sdk/Trace/Tracer.php (hunk 2 above)
Comment (zsistla) [reply sdk/Trace/Tracer.php:173]: Trying to ensure the immutableness of resource.  The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/sdk.md) says `A Resource is an immutable representation of the entity producing telemetry.` Similar to how we handle the immutableness of `context` in this [getter](https://github.com/open-telemetry/opentelemetry-php/blob/master/sdk/Trace/Span.php#L67).

Comment (bobstrecansky) [reply sdk/Trace/Tracer.php:173]: Ahh perfect.  Thanks for the clarification!

**Pull request merged**